docs/docs/vm-fundamentals/hints.md

```diff
@@ -2,6 +2,138 @@
 sidebar_position: 3
 ---
 
-# Hints
+# Hint
+  ## HINTS  IN CAIRO VIRTUAL MACHINE
+Hints in Cairo are pieces of Python code only executed by the sequencer and are not included in the bytecode. They instruct the prover on how to handle nondeterministic instructions. These nondeterministic instructions are programs that have different outcomes at execution.   
+Hints are guides that developers can provide to the Cairo virtual machine to optimize execution or provide additional information about the program. These hints help improve performance or help the VM make better decisions during the execution of programs. 
```

Comment:
This part is partially correct, I understand is a complex subject and that you want to explain the "why" of the hints. Let's reduce this paragraph to something simpler: "Hints are used to help the Virtual Machine determine the next path of execution it should take"